### PR TITLE
Add section about Workload's requeueState field to KEP#349

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind documentation
/kind cleanup

#### What this PR does / why we need it:
Adds section about Workload's requeueState field to KEP#349. There is no new changes, it only updates KEP to better reflect current state

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
Thank you for the update which is just reflecting the current API.

File: keps/349-all-or-nothing/README.md [hunk 1]
```diff
@@ -279,6 +280,37 @@ const (
 
 ```
 
+### Workload API changes
```
Comment (tenzen-y) [inline keps/349-all-or-nothing/README.md:283]: This API was not introduced by the All or nothing feature.
This is introduced by https://github.com/kubernetes-sigs/kueue/tree/main/keps/1282-pods-ready-requeue-strategy#workload.

Comment (tenzen-y) [inline keps/349-all-or-nothing/README.md:283]: This API was not introduced by the All or nothing feature.
This is introduced by https://github.com/kubernetes-sigs/kueue/tree/main/keps/1282-pods-ready-requeue-strategy#workload.

Comment (tenzen-y) [reply keps/349-all-or-nothing/README.md:283]: I opened the reverting PR: https://github.com/kubernetes-sigs/kueue/pull/3917

**Pull request merged**